baybe/acquisition/_builder.py

```diff
@@ -92,6 +100,12 @@ class BotorchAcquisitionFunctionBuilder:
 
     def __attrs_post_init__(self) -> None:
         """Initialize the building process."""
+        # Use provided device or get default
+        if self.device is None:
+            self.device = getattr(self.surrogate, "device", None)
+            if self.device is None and torch.cuda.is_available():
+                self.device = get_default_device()
```

Comment:
Since this is purely related to the device attribute, I think this better belongs into a dedicated `attrs default` method (i.e. using `@device.default`)